lib/chef/resource/apt_repository.rb

```diff
@@ -0,0 +1,47 @@
+#
+# Author:: Thom May (<thom@chef.io>)
+# Copyright:: Copyright (c) 2016 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "chef/resource"
+
+class Chef
+  class Resource
+    class AptRepository < Chef::Resource
+      resource_name :apt_repository
+      provides :apt_repository
+
+      property :repo_name, String, name_property: true
+      property :uri, String
+      property :distribution, String, default: lazy { node["lsb"]["codename"] }
```

Comment:
I love that this was added to core Chef, but...

@thommay Having this as a default is not a good idea because some Apt repos do not use a distribution and having this here makes it impossible to use those things. Also, it is a different behavior than the existing `apt` cookbook, which causes it to break backwards compatibility.